accepted/2020/net5/net5.md

```diff
@@ -397,6 +397,167 @@ MSBuild too._
+this:
+
+```xml
+<ItemGroup Condition="'$(TargetFramework)' >= 'net5.0'`">
+```
+
+but this doesn't work because that would be a string comparison, not a version
+comparison. Instead, the user has to write it like this:
+
+```xml
+<ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(TargetFrameworkVersion)' >= '3.0'">
+```
+
+This works for conditions on item groups because they are evaluated after
+properties. Since the user's project file defines the `TargetFramework`
+property, the SDK logic that expands it into the other properties such as
+`TargetFrameworkIdentifier` and `TargetFrameworkVersion` has to live in
+`*.targets`, i.e. at the bottom of the project file. That means these
+automatically expanded properties aren't available for the user when defining
+other properties. This happens to work for items because items are evaluated
+after all properties are evaluated.
+
+Concretely, this means the user cannot define properties like this:
+
+```XML
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;netcoreapp3.1</TargetFramework>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETStandard'">
+    <SomeProperty>Some .NET Standard specific value<SomeProperty>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
+    <SomeProperty>Some .NET Core specific value<SomeProperty>
+  </PropertyGroup>
+
+</Project>
+```
+
+In the past, we've seen people working this around by using string processing
+functions against the `TargetFramework` property, which is less than ideal.
+
+Ideally, we'd expose functionality such that the user can do version checks:
+
+```XML
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;netcoreapp3.1</TargetFramework>
+  </PropertyGroup>
+
+  <PropertyGroup TargetFramework="netstandard">
+    <SomeProperty>Some value that applies to all versions of .NET Standard<SomeProperty>
+  </PropertyGroup>
+
+  <PropertyGroup TargetFramework=">=netcoreapp2.0">
+    <SomeProperty>Some value that applies to .NET Core 2.0 and later<SomeProperty>
+  </PropertyGroup>
+
+  <PropertyGroup TargetFramework="==net5.0-ios13.0">
+    <SomeProperty>Some value that only applies to .NET 5 + iOS 13.0<SomeProperty>
+  </PropertyGroup>
+
+  <PropertyGroup TargetPlatform="windows">
+    <SomeProperty>Some value that applies to all version of Windows<SomeProperty>
+  </PropertyGroup>
+
+  <PropertyGroup TargetPlatform=">=ios-12.0">
+    <SomeProperty>Some value that applies to iOS 12.0 and later<SomeProperty>
+  </PropertyGroup>
+
+</Project>
+```
+
+The idea is:
+
+* Property groups, properties, and item groups get new attributes
+  `TargetFramework` and `TargetPlatform`.
+* The value can be prefixed with an optional conditional operator `==`, `!=`,
+  `<`, `<=`, `>`, and `>=`. If the operator is omitted, `==` is assumed.
```

Comment:
Are these conditions sufficient, or is there a need for something like `IsCompatibleWith`?

Reply:
My thinking was that that's what `TargetFramework=">=net5.0"` will basically do.

Reply:
Moved to: https://github.com/microsoft/msbuild/issues/5171